sklearn/metrics/_classification.py

```diff
@@ -2013,63 +2038,110 @@ class after being classified as negative. This is the case when the
     >>> class_likelihood_ratios(y_true, y_pred, labels=["non-cat", "cat"])
     (1.5, 0.75)
     """
-
+    # TODO(1.8): When `raise_warning` is removed, the default return value with
+    # zero_division="warn" does not need to be "np.nan" anymore and should be the lowest
+    # score for each metric respectively (1 for LR+ and 0 for LR-) to match the other
+    # functions that take a zero_division param. Return values and warning messages need
+    # to be updated. The Warns section in the docstring needs be be re-written and
+    # the hidden option for zero_division ("default") be removed and the default set to
+    # "warn" to match the other functions that take a zero_division param.
     y_type, y_true, y_pred = _check_targets(y_true, y_pred)
     if y_type != "binary":
         raise ValueError(
             "class_likelihood_ratios only supports binary classification "
             f"problems, got targets of type: {y_type}"
         )
 
+    if raise_warning != "deprecated":
+        if zero_division != "default":
+            raise ValueError(
+                "`zero_division` and `raise_warning` cannot both be set. "
+                "Use either one to control the return values in case "
```

Comment:
We should advocate only `zero_division` here since `raise_warning` is going to be removed and raise a subsequent warning if used.